http/src/http.ml

```diff
@@ -755,6 +755,10 @@ module Request = struct
     | `GET | `HEAD | `CONNECT | `TRACE -> `No
     | `DELETE | `POST | `PUT | `PATCH | `OPTIONS | `Other _ ->
         Transfer.has_body req.encoding
+
+  let[@warning "-3"] make ?(meth = `GET) ?(version = `HTTP_1_1) 
+      ?(headers = Header.empty) ?scheme resource =
+    { headers; meth; scheme; resource; version; encoding = Transfer.Unknown }
```

Comment:
I would still keep the encoding and document it as in `Response.Header`, we will remove it once we remove the field:

```
has an encoding value
      determined from the content of [headers] or if no proper header is
      present, using the value of [encoding]. 
```

Reply:
Just so that we are on the same page, this is what I am talking about: https://github.com/mirage/ocaml-cohttp/blob/b2c3198658fda65cab7f5275bcd2ea329a146b9f/http/src/http.mli#L417-L429
I think it is more consistent if we do the same thing in both places, and we should let the deprecation stay until the next next version